scripts/plugins/firewall-port

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+#
+# Copyright (c) Citrix Systems 2017. All rights reserved.
+
+set -e
+
+# Using this script to open/clear port. 
+
+
+OP="$1"
+PORT="$2"
+CHAIN="RH-Firewall-2-INPUT"
+RULE="-p tcp -m conntrack --ctstate NEW -m tcp --dport $PORT -j ACCEPT"
+
+case "${OP}" in
+    open)
+        if iptables -S | grep $CHAIN | grep -e "${RULE}" > /dev/null
```

Comment:
Do not grep for CHAIN.
Specify it as a parameter to the command:
`iptables -S "${CHAIN}"` will print only the information about the specified chain.

If you were to use `grep` to search for the specific rule then to be on the safe side, you should use the `-F` and `-x` options to grep, to match the whole line and avoid interpreting our search-strings as regular expressions:
`iptables -S "${CHAIN"}" 2>&1 | grep -Fx "-A ${CHAIN} ${RULE}" > /dev/null`

But using `grep` is a bad idea.
Read the man-page for iptables:
Type `man iptables`.
Note this part:
```
   COMMANDS
       These  options specify the desired action to perform. Only one of them
       can be specified on the command line unless  otherwise  stated  below.
       For  long  versions  of  the command and option names, you need to use
       only enough letters to ensure that iptables can differentiate it  from
       all other options.
       [...snip...]

       -C, --check chain rule-specification
              Check whether a rule matching the specification does  exist  in
              the  selected  chain. This command uses the same logic as -D to
              find a matching entry, but does not alter the existing iptables
              configuration  and  uses  its  exit code to indicate success or
              failure.
```
Use that if you want to check for the presence of a rule.